tests/std/tests/GH_003570_allocate_at_least/test.cpp

```diff
@@ -0,0 +1,104 @@
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include <cassert>
+#include <deque>
+#include <memory>
+#include <sstream>
+#include <string>
+#include <syncstream>
+#include <utility>
+#include <vector>
+
+using namespace std;
+
+struct {
+    void assert_set() {
+        assert(consume());
+    }

```

Comment:
Suggestion: I find the naming of this method to be confusing. I think it would be simpler if it were eliminated, and callers directly asserted.